src/main/scala/org/camunda/feel/FeelEngine.scala

```diff
@@ -161,6 +161,11 @@ class FeelEngine(
           failure = Failure(s"failed to evaluate expression '${exp.text}': $cause"),
           suppressedFailures = context.failureCollector.failures
         )
+      case _ if containsAssertionError(context) =>
```

Comment:
:x: We should move this case to the top to ensure that we produce always the same failure message.

Currently, the failure message is different if the `assert()` function is wrapped inside another function or not.